specs/_features/eip7732/beacon-chain.md

```diff
@@ -0,0 +1,796 @@
+    - [`is_valid_indexed_payload_attestation`](#is_valid_indexed_payload_attestation)
+    - [`is_parent_block_full`](#is_parent_block_full)
+  - [Beacon State accessors](#beacon-state-accessors)
+    - [`get_ptc`](#get_ptc)
+    - [Modified `get_attesting_indices`](#modified-get_attesting_indices)
+    - [`get_payload_attesting_indices`](#get_payload_attesting_indices)
+    - [`get_indexed_payload_attestation`](#get_indexed_payload_attestation)
+- [Beacon chain state transition function](#beacon-chain-state-transition-function)
+  - [Block processing](#block-processing)
+    - [Withdrawals](#withdrawals)
+      - [Modified `process_withdrawals`](#modified-process_withdrawals)
+    - [Execution payload header](#execution-payload-header)
+      - [New `verify_execution_payload_header_signature`](#new-verify_execution_payload_header_signature)
+      - [New `process_execution_payload_header`](#new-process_execution_payload_header)
+    - [Operations](#operations)
+      - [Modified `process_operations`](#modified-process_operations)
+      - [Payload Attestations](#payload-attestations)
+        - [`process_payload_attestation`](#process_payload_attestation)
+    - [Modified `process_execution_payload`](#modified-process_execution_payload)
+      - [New `verify_execution_payload_envelope_signature`](#new-verify_execution_payload_envelope_signature)
+    - [Modified `is_merge_transition_complete`](#modified-is_merge_transition_complete)
+    - [Modified `validate_merge_block`](#modified-validate_merge_block)
+- [Testing](#testing)
+  - [Modified `initialize_beacon_state_from_eth1`](#modified-initialize_beacon_state_from_eth1)
+  - [Modified `is_merge_transition_complete`](#modified-is_merge_transition_complete-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification of the enshrined proposer builder separation feature. 
+
+*Note:* This specification is built upon [Electra](../../electra/beacon-chain.md) and is under active development.
+
+This feature adds new staked consensus participants called *Builders* and new honest validators duties called *payload timeliness attestations*. The slot is divided in **four** intervals. Honest validators gather *signed bids* (a `SignedExecutionPayloadHeader`) from builders and submit their consensus blocks (a `SignedBeaconBlock`) including these bids at the beginning of the slot. At the start of the second interval, honest validators submit attestations just as they do previous to this feature). At the  start of the third interval, aggregators aggregate these attestations and the builder broadcasts either a full payload or a message indicating that they are withholding the payload (a `SignedExecutionPayloadEnvelope`). At the start of the fourth interval, some validators selected to be members of the new **Payload Timeliness Committee** (PTC) attest to the presence and timeliness of the builder's payload.
```

Comment:
I think `(Signed|)ExecutionPayloadHeader` should be renamed to `(Signed|)ExecutionPayload(Bid|Commitment|Info|Metadata)`, as it serves a new purpose.

`.+Header` so far is used for SSZ summaries of the corresponding base type, which is no longer the case as the full `ExecutionPayload` still contains the extra fields such as `transactions_root` and so on.

Also, the `ExecutionPayloadHeader` is used by light clients as part of `LightClientHeader`. They still need the previous semantics of it being the SSZ summary of the `ExecutionPayload`. Therefore, recommending to drop `ExecutionPayloadHeader` from the `beacon-chain.md` spec and creating a new type of different name.

Reply:
Yeah @ethDreamer had the same complaint, I don't mind any name change eventually for this object. It's not a trivial change to implement in the spec, so probably will wait until this is agreed to be shipped.